design-documents/testing/functional/mftf-stack-configuration.md

```diff
@@ -0,0 +1,29 @@
+# Overview
+
+Technically, Magento Functional Testing Framework has two types of stack dependencies:
+1. Responsible for maintaining tests (PHP, Composer, Allure)
+2. Setup that we run test on (Selenium, ChromeDriver, GeckoDriver, Phantom) and it's dependencies (Java)
+
+## Current State
+
+Current MFTF codebase requires to keep whole stack at single environment. Either dependencies necessary to maintain tests and the webdriver stack on which we run tests needs to be located at single machine. MFTF expects dependencies to be available by running local shell command (eg. `java -version 2>&1`).
+
+Only ChromeDriver is supported at the moment.
+
+## Problem
+
+Current approach forces Developers to use single, monolith testing environment. It also couples MFTF with very specific stack, limiting it's further flexibility to use different web drivers.
+
+As an example - Allure can be easily dockerized and used as an external service, MFTF does not support this setup.
```

Comment:
Can you please provide some more examples of decoupling that you need?

Reply:
@tomreece As far as I know @lbajsarowicz wants to be able to specify own configuration for his new Magento Extension/Module and he wants to have an extension point for configuring this. This leads us to introduce all ENV configurations via XML.